core/src/main/java/org/apache/iceberg/actions/SizeBasedDataRewriter.java

```diff
@@ -45,6 +47,7 @@ public abstract class SizeBasedDataRewriter extends SizeBasedFileRewriter<FileSc
   public static final String DELETE_FILE_THRESHOLD = "delete-file-threshold";
 
   public static final int DELETE_FILE_THRESHOLD_DEFAULT = Integer.MAX_VALUE;
+  private static final double DELETE_RATIO_THRESHOLD = 0.3;
```

Comment:
Let's add an option to configure this in a separate PR.

Reply:
opened https://github.com/apache/iceberg/issues/12081 to make this configurable